### ISSUE TITLE
Browserstack Credentials Not Being passed to DriverInitialiser

### ISSUE THREAD

**Issue by tomrobinsonv** [SpecFlowOSS/SpecFlow.Actions#88]
I have set up my specflow project up almost identically to match your Browserstack implementation, although the only thing I haven't been able to get working is the credentials. 

SpecFlow.NUnit - V3.9.74
SpecFlow.Actions.Browserstack - V0.1.322

Target Framework - netcoreapp3.1
Language Version - 9.0

The error I'm getting now

`
Message: 
    OpenQA.Selenium.WebDriverException : Unexpected error. Authorization required
    TearDown : OpenQA.Selenium.WebDriverException : Unexpected error. Authorization required

  Stack Trace: 
    RemoteWebDriver.UnpackAndThrowOnError(Response errorResponse)
    RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
    RemoteWebDriver.StartSession(ICapabilities desiredCapabilities)
    RemoteWebDriver.ctor(ICommandExecutor commandExecutor, ICapabilities desiredCapabilities)
    RemoteWebDriver.ctor(Uri remoteAddress, ICapabilities desiredCapabilities, TimeSpan commandTimeout)
    RemoteWebDriver.ctor(Uri remoteAddress, ICapabilities desiredCapabilities)
    RemoteWebDriver.ctor(Uri remoteAddress, DriverOptions options)
    BrowserstackDriverInitialiser.GetWebDriver(DriverOptions options)
    BrowserstackChromeDriverInitialiser.CreateWebDriver(ChromeOptions options)
    DriverInitialiser`1.Initialise()
    <29 more frames...>
    Lazy`1.get_Value()
    BrowserDriver.get_Current()
    BrowserstackRuntimePlugin.RuntimePluginTestExecutionLifecycleEventEmitter_AfterScenario(Object sender, RuntimePluginAfterScenarioEventArgs e)
    RuntimePluginTestExecutionLifecycleEvents.RaiseAfterScenario(IObjectContainer objectContainer)
    RuntimePluginTestExecutionLifecycleEventEmitter.RaiseExecutionLifecycleEvent(HookType hookType, IObjectContainer container)
    TestExecutionEngine.FireEvents(HookType hookType)
    TestExecutionEngine.FireScenarioEvents(HookType bindingEvent)
    TestExecutionEngine.OnScenarioEnd()
    TestRunner.OnScenarioEnd()
    ManualVehicleSearchFeature.TestTearDown()
`

My BrowserstackCredentialProvider class
`
 class BrowserstackCredentialProvider : ICredentialProvider
    {
        private readonly Lazy<string?> _browserstackUsername;
        private readonly Lazy<string?> _accessKey;

        public BrowserstackCredentialProvider()
        {
            Console.WriteLine("FEtching authoria");
            _browserstackUsername = new Lazy<string?>(() => Environment.GetEnvironmentVariable("BROWSERSTACK_USER"));
            _accessKey = new Lazy<string?>(() => Environment.GetEnvironmentVariable("BROWSERSTACK_KEY"));
        }

        public string? Username => _browserstackUsername.Value;
        public string? UsernameArgumentName => "browserstack.user";
        public string? AccessKey => _accessKey.Value;
        public string? AccessKeyArgumentName => "browserstack.key";
    }
`

I have double checked my environment variables and they are definitely valid. I also added the logging line in the constructor method but this never gets called.

**Comment by mrampass** [comment]
Hello , Could you please mention the webdriver package version that you are using ?

**Comment by tomrobinsonv** [comment]
Hii, the Selenium Webdriver version I have is 3.141.0

**Comment by tomrobinsonv** [comment]
I've just updated to the latest version 4.1.1 but still the same error.

**Comment by tomrobinsonv** [comment]
Hi, quick update. I have managed to access the browserstack server but only after setting my credentials in the specflow.actions.json file which will be exposed in the repo. Even though this is now working, I have ran into another issue which is that the driver is not being destroyed after a scenario. Every taste hangs for over a minute after the test has finished and manually quitting the driver in the `AfterScenario` hook method doesn't work.
